front/src/applications/operationalStudies/hooks/useSimulationResults.ts

```diff
@@ -1,164 +1,128 @@
-  extractPacedTrainIdFromOccurrenceId,
-} from 'utils/trainId';
 
 import type { SimulationResults } from '../types';
+import { preparePathPropertiesData } from '../utils';
+
+const PREFER_CACHE_VALUE = true;
 
 /**
  * Prepare data to be used in simulation results
  */
 const useSimulationResults = (infraId: number): SimulationResults | undefined => {
+  const { t } = useTranslation('operational-studies');
+
   const electricalProfileSetId = useSelector(getOperationalStudiesElectricalProfileSetId);
   const selectedTrainId = useSelector(getSelectedTrainId);
 
-  const editoastSelectedTrainId = useMemo(() => {
-    if (!selectedTrainId) return undefined;
-    if (isTrainScheduleId(selectedTrainId)) {
-      return extractEditoastIdFromTrainScheduleId(selectedTrainId);
-    }
-    const pacedTrainId = extractPacedTrainIdFromOccurrenceId(selectedTrainId);
-    return extractEditoastIdFromPacedTrainId(pacedTrainId);
-  }, [selectedTrainId]);
+  const [simulationResults, setSimulationResults] = useState<SimulationResults>();
 
-  const { data: selectedTrainSchedule } = osrdEditoastApi.endpoints.getTrainScheduleById.useQuery(
-    {
-      id: editoastSelectedTrainId!,
-    },
-    { skip: !editoastSelectedTrainId || (selectedTrainId && !isTrainScheduleId(selectedTrainId)) }
-  );
+  const train = useSelectedTrain();
 
-  const { data: selectedPacedTrain } = osrdEditoastApi.endpoints.getPacedTrainById.useQuery(
+  const { data: path } = osrdEditoastApi.endpoints.getTrainPath.useQuery(
     {
-      id: editoastSelectedTrainId!,
+      id: selectedTrainId!,
+      infraId,
     },
-    { skip: !editoastSelectedTrainId || (selectedTrainId && !isOccurrenceId(selectedTrainId)) }
+    {
+      skip: !selectedTrainId,
+    }
   );
 
-  const { data: rawTrainSchedulePath } =
-    osrdEditoastApi.endpoints.getTrainScheduleByIdPath.useQuery(
-      {
-        id: editoastSelectedTrainId!,
-        infraId,
-      },
-      {
-        skip: !editoastSelectedTrainId || (selectedTrainId && !isTrainScheduleId(selectedTrainId)),
-      }
-    );
-
-  const { data: rawPacedTrainPath } = osrdEditoastApi.endpoints.getPacedTrainByIdPath.useQuery(
+  const { data: simulation } = osrdEditoastApi.endpoints.getTrainSimulation.useQuery(
     {
-      id: editoastSelectedTrainId!,
+      id: selectedTrainId!,
       infraId,
+      electricalProfileSetId,
     },
     {
-      skip: !editoastSelectedTrainId || (selectedTrainId && !isOccurrenceId(selectedTrainId)),
+      skip: !selectedTrainId,
     }
   );
-  const path = useMemo(() => {
-    if (!selectedTrainId) return undefined;
 
-    if (isTrainScheduleId(selectedTrainId)) {
-      return rawTrainSchedulePath?.status === 'success' ? rawTrainSchedulePath : undefined;
+  const [postPathProperties] =
+    osrdEditoastApi.endpoints.postInfraByInfraIdPathProperties.useLazyQuery();
+  const [getRollingStockByName] =
+    osrdEditoastApi.endpoints.getRollingStockNameByRollingStockName.useLazyQuery();
+
+  useEffect(() => {
```

Comment:
Can we avoid `useEffect()` here? See: https://react.dev/learn/you-might-not-need-an-effect